### PR TITLE
Consider fees tracking for block execution weight and set `AvailableBlockRatio` to be 1

### PR DESCRIPTION
**Pull request by lovesh**

### CODE REVIEW

**Comment by bddap** [comment]
Our blocks always have one extrinsic in them. Is that an operational extrinsic?

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -189,6 +202,9 @@ impl system::Trait for Runtime {
     type BlockExecutionWeight = BlockExecutionWeight;
     /// The base weight of any extrinsic processed by the runtime, independent of the
     /// logic of that extrinsic. (Signature verification, nonce increment, fee, etc...)
+    /// The storage item `TxnFees` would potentially be read and written after each extrinsic if that
+    /// pays fees but that read and write goes to DB only once per block due to Substrate's Substrate's _overlay change set_
```
Comment (bddap) [inline runtime/src/lib.rs:206]: Nit
```suggestion
    /// pays fees but that read and write goes to DB only once per block due to Substrate's _overlay change set_
```

**Comment by lovesh** [comment]
> Our blocks always have one extrinsic in them. Is that an operational extrinsic?

Which one?

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -148,9 +151,19 @@ parameter_types! {
     pub const BlockHashCount: BlockNumber = 2400;
     /// We allow for 1 seconds of compute with a 3 second average block time.
     pub const MaximumBlockWeight: Weight = WEIGHT_PER_SECOND;
-    pub const AvailableBlockRatio: Perbill = Perbill::from_percent(75);
+    /// Since there are no `Operational` transactions as of now, the whole block can be filled with
+    /// `Noraml` transactions.
```
Comment (bddap) [inline runtime/src/lib.rs:155]: Nit
```suggestion
    /// `Normal` transactions.
```

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -148,9 +151,19 @@ parameter_types! {
...
     /// Assume 10% of weight for average on_initialize calls.
     pub MaximumExtrinsicWeight: Weight = AvailableBlockRatio::get().saturating_sub(Perbill::from_percent(10)) * MaximumBlockWeight::get();
+    /// DefaultBlockExecutionWeight is the weight of any empty block.
+    /// After each block we
+    /// - update stats, which is 1 read and 1 write
+    /// - check if there is any fees in storage item `TxnFees`, which is 1 read
+    /// - credit fees to block author's account which is 1 read and 1 write
+    /// - reset the storage item `TxnFees`, 1 write
+    /// Thus in the worst case, we do 3 reads and 3 writes
+    pub BlockExecutionWeight: Weight = DefaultBlockExecutionWeight::get() + RocksDbWeight::get().reads_writes(3, 3);
```
Comment (bddap) [inline runtime/src/lib.rs:166]: Nit
```suggestion
    pub BlockExecutionWeight: Weight = DefaultBlockExecutionWeight::get() +
        <Runtime as system::Trait>::DbWeight::get().reads_writes(3, 3);
```

**Pull request merged**